lib/data-types.js

```diff
@@ -302,6 +346,13 @@ DECIMAL.prototype.toSql = function() {
 
   return 'DECIMAL';
 };
+DECIMAL.prototype.validate = function(value) {
+  if (!Validator.isNumeric(value)) {
```

Comment:
Should be isFloat (isNumeric is used to check for a string only containing nums)